r1cs/gadgets/crypto/src/crh/poseidon/mod.rs

```diff
@@ -158,7 +135,7 @@ impl<
         {
             // Apply the S-BOX to each of the elements of the state vector
             for (j, d) in state.iter_mut().enumerate() {
-                Self::mod_inv_sbox(cs.ns(|| format!("mod_inv_S-Box_4_{}_{}", P::R_F-1, j)), d)?;
+                SBG::apply(cs.ns(|| format!("mod_inv_S-Box_4_{}_{}", P::R_F-1, j)), d)?;
```

Comment:
As stated before, the last round is not treated differently than the other full rounds.